### PR TITLE
Redhat systemd location

### PR DESCRIPTION
**Pull request by gdanko**
debian and redhat use different paths for their systemd unit files. This patch does two things. 1) determine whether the distro is redhat-based or debian-based and set the systemd path accordingly. 2) Verify this path exists before attempting to write the unit file.

### CODE REVIEW

**Comment by gdanko** [comment]
@lwander @jtk54 Have a look, guys.

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy [hunk 1]
```diff
@@ -79,11 +79,27 @@ class SpinnakerPackagePlugin implements Plugin<Project> {
...
+        */
+        String systemdPath
+        def apt = 'which apt'.execute()
+        def yum = 'which yum'.execute()
+        apt.waitFor()
+        yum.waitFor()
+        if (apt.exitValue() == 0 && yum.exitValue() == 1) {
+            systemdPath = '/lib/systemd/system'
+        } else if (apt.exitValue() == 1 && yum.exitValue() == 0) {
+            systemdPath = '/usr/lib/systemd/system'
```
Comment (lwander) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: What if neither is true? Or both? Also seems systemdPath lives only in the scope of these if statments, how is it accessed below?

Comment (lwander) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: Ah you declare it above, but there is still the chance for NPE below

File: src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy [hunk 2]
```diff
@@ -79,11 +79,27 @@ class SpinnakerPackagePlugin implements Plugin<Project> {
                 setFileType(new Directive(Directive.RPMFILE_CONFIG | Directive.RPMFILE_NOREPLACE))
             }
         }
-        
+
+        /* Determine the distro and set the path for systemd unit files
+           if apt exits 0 and yum exits 1 then this is a debian-based
+           OS. Conversely, if apt exits 1 and yum exits 0 it is a
+           redhat-based OS.
```
Comment (lwander) [inline src/main/groovy/com/netflix/spinnaker/gradle/ospackage/SpinnakerPackagePlugin.groovy]: Isn't this run when the package is built? Does that mean that we need to build RPMs on redhat, and debs on debian?

**Comment by gdanko** [comment]
This is not going to work.

**Pull request closed without merging**